### ISSUE TITLE
BUG: try-catch with window object will disappear

### ISSUE THREAD

**Issue by SaekiRaku** [terser/terser#506]
**Bug report or Feature request?**

Bug Report

**Version**

Using `https://try.terser.org`, guessing it's `v4.3.9` by now?

**`terser` input**

```
try {
    window;
} catch(e) {
    console.log(e);
}
```

**`terser` output or error**

output nothing.

**Expected result**

```
try {
    window;
} catch(e) {
    console.log(e);
}
```

I'm using this code fragment inside a `WebWorker` js file. The 'try-catch' for `window` object, can let the code know if it running inside a worker, and then do different behaviour. I'm trying not to make this issue involved other libraries, therefor a detailed example is provided here: https://github.com/parcel-bundler/parcel/issues/3748

BTW, right now I can change `window` to `window.document` to get expected result from `terser`.

**Comment by fabiosantoscode** [comment]
Have you enabled strict mode in the script?

**Comment by SaekiRaku** [comment]
@fabiosantoscode 

No, but I've tried again with `strict` mode enabled. The `try-catch` block is still disappeared.

I've looked around the source code, and find out maybe it's because of a line only have `window` or something else inside it. Terser will consider it as an `AST_SimpleStatement`? And then if the string of that line coincidentally exists in `pure_prop_access_globals`, tarser will return an `AST_EmptyStatement` node. I'm not sure about this by now, but I guess terser doesn't consider the situation that the `Statement` is inside a `try-catch` block.

```
// lib/compress/index.js
// Line of 4378

pure_prop_access_globals = [ "Number", "String", "window", ...etc ];

...

def_optimize(AST_SimpleStatement, function(self, compressor) {
    if (self.body instanceof AST_SymbolRef && pure_prop_access_globals.indexOf(self.body.name) !== -1) {
        return make_node(AST_EmptyStatement, self);
    }
    ...
}
```
